lib/jekyll/document.rb

```diff
@@ -123,13 +123,34 @@ def cleaned_relative_path
         relative_path[0..-extname.length - 1].sub(collection.relative_directory, "")
     end
 
+    # Returns an array of sub directories that represents the document's path
+    # relative to the collection directory
+    def subdirs
+      cleaned_relative_path.split(File::SEPARATOR)
+        .reject { |c| c.empty? || c.eql?(basename_without_ext) }
+    end
+
     # Determine whether the document is a YAML file.
     #
     # Returns true if the extname is either .yml or .yaml, false otherwise.
     def yaml_file?
       %w(.yaml .yml).include?(extname)
     end
 
+    # Determine whether the document is a JSON file.
+    #
+    # Returns true if extname == .json, false otherwise.
+    def json_file?
```

Comment:
I'm wondering if this should really exist or exist in its current form. YAML files are very close to JSON files and will be processed the same way. I am wondering if `json_file?` should include ` || yaml_file?` so any place it is used can be reduced since they're always side by side.